start_cerberus.py

```diff
@@ -94,6 +99,22 @@ def main(cfg):
         while (int(iteration) < iterations):
             iteration += 1
 
+            # Check if api server url is ok
+            server_status = kubecli.is_url_available(api_server_url)
+            if not server_status:
+                # Start time only if the api server ok flag is being set for the first time to false
+                if api_server_ok:
+                    start_time = datetime.datetime.now()
+                    api_server_ok = False
+
+            else:
+                if not api_server_ok:
+                    end_time = datetime.datetime.now()
+                    # Log the time the api server has been down for
+                    logging.info("Api server was down for %s" % (str(end_time - start_time)))
```

Comment:
@yashashreesuresh bought up a good point, so @paigerube14 we won't be able to capture the exact time duration of the API server unavailability as there's a sleep duration between each duration. For example, if the apiserver is reported as not healthy in iteration 1 with sleep duration set to say 5 mins and the apiserver recovered within a min, the time duration will take the sleep duration into account as well as the next check happens after the sleep. 

Think it's fine to just expose the no-go signal without displaying the down time as it's tricky. Also if it's unhealthy for just few seconds, cerberus will miss it like you mentioned unless the sleep time is set to a low value like 2 seconds. We might want to add a note that the duration needs to be set to a lower value for catching things which flap vs long term failures, this will increase the number of API requests but it's fine if the user wants to capture the flaps or we need to find a way to capture what happened during the sleep time similar to https://github.com/openshift-scale/cerberus/pull/52 ( might not be possible for all the cases I guess unless there's a way to query the apiserver health condition given a start and stop time). Thoughts?

Reply:
@mffiedler What are your thoughts here on not displaying the down time as it is not exact because of the sleep duration between iterations? 
 
I can add a comment to the readme that we can lower the duration to see a more specific downtime for this api. I'm really not sure how to query the apiserver for a specific start or stop time, seems like in the linked issue they were able to look at the timestamp and I'm not aware of the apiserver having this information. Let me know if that is not the case

Reply:
I am ok with not showing down time - it would still be nice to keep a count of how many times the API is unavailable.   Since it is transient, the no-go signal might be for a single iteration.   I'm interested in whether we get repeated failures.

Reply:
Agreed, we can keep a track of the count and eventually expose it as a metric in addition to the no/no-go signal. This way the user/other tools will be able to consume the signal at that instance of time and use the metrics collected to understand what happened given a time range like mentioned in https://github.com/openshift-scale/cerberus/issues/58.

Reply:
It’s good to display the down time only if we know the exact duration. We can keep track of the number of times apiserver goes down by checking the apiserver status in each iteration. Though we aren’t checking the number of times the apiserver went down during the sleep interval, this would be a good approximate count.

Reply:
I have added a counter to increase each time the api server goes down. For now I am not doing anything with it. For issue #58 we could incorporate this into a json format to keep track of all the metrics we want. 
Should this check be part of the go/no-go signal?